src/plugins/WebOfTrust/util/DelayedBackgroundJob.java

```diff
@@ -0,0 +1,270 @@
+package plugins.WebOfTrust.util;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import freenet.node.FastRunnable;
+import freenet.node.PrioRunnable;
+import freenet.support.Executor;
+import freenet.support.Ticker;
+import freenet.support.io.NativeThread;
+
+/**
+ * A {@link BackgroundJob} that is executed after triggers have been aggregated for a given delay.
+ *
+ * @author bertm
+ * @see BackgroundJob
+ * @see DelayedBackgroundJobFactory
+ */
+public class DelayedBackgroundJob implements BackgroundJob {
+    static enum JobState {
+        /** Waiting for a trigger, no running job thread or scheduled job. */
+        IDLE,
+        /** Waiting for the scheduled job to be executed, no running job thread. */
+        WAITING,
+        /** Running the job in the job thread. */
+        RUNNING,
+        /** Waiting for the running job thread to finish before we terminate. */
+        TERMINATING,
+        /** Terminated, no running job thread. */
+        TERMINATED
+    }
+
+    /** Special time value to indicate the absence of an oldest trigger. */
+    private final static long NO_TRIGGER = 0;
```

Comment:
To speed up the release, I've fixed this myself in commit d7b8705f9d289c9e327397fd6999841e79f1526a

-> Done